Cross-compilation failure on OS X host

This issue has been mentioned here: https://github.com/hashicorp/nomad/pull/438#issuecomment-157568019

```
$ go version
go version go1.5.1 darwin/amd64

$ make updatedeps
--> Installing build dependencies
--> Updating build dependencies

$ make bin
/Users/legal/go/src/github.com/hashicorp/nomad
==> Getting dependencies...
==> Removing old directory...
==> Building...
Number of parallel builds: 4

-->   windows/amd64: github.com/hashicorp/nomad
-->     linux/amd64: github.com/hashicorp/nomad
-->       linux/386: github.com/hashicorp/nomad
-->    darwin/amd64: github.com/hashicorp/nomad
-->     windows/386: github.com/hashicorp/nomad

2 errors occurred:
--> linux/amd64 error: exit status 2
Stderr: # runtime/cgo
ld: unknown option: --build-id=none
clang: error: linker command failed with exit code 1 (use -v to see invocation)

--> linux/386 error: exit status 2
Stderr: # runtime/cgo
ld: unknown option: --build-id=none
clang: error: linker command failed with exit code 1 (use -v to see invocation)

make: *** [bin] Error 1
```

I'm new with Golang :(. Please, advice - how can I fix that? I believe, it should be possible to cross-compile Nomad for Linux. 

P.s. Seems like there is some new stuff in Go 1.5, related to cross-compilation:
http://dave.cheney.net/2015/03/03/cross-compilation-just-got-a-whole-lot-better-in-go-1-5
Is it the reason of this issue?

cc: @cbednarski

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -44,7 +44,6 @@ gox \
     -arch="${XC_ARCH}" \
     -osarch="!linux/arm !darwin/386" \
     -ldflags "-X main.GitCommit ${GIT_COMMIT}${GIT_DIRTY}" \
-    -cgo \
     -output "pkg/{{.OS}}_{{.Arch}}/nomad" \
     .
 

